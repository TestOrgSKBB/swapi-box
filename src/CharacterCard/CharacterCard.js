import React, { Component } from 'react';
import './CharacterCard.scss'
import favoriteImg from '../images/favorite.png';
import activeFavoriteImg from '../images/active-favorite.png';
import PropTypes from 'prop-types';

class Character extends Component {

  toggleFavorited = () => {
    const { name } = this.props.character;
    this.props.updateFavorite(name);
  }

  returnFilms = () => {
    const films = this.props.character.films;
    return films.map((film, i) => <p className="character__card--films" key={i}>{film.title}</p>);
  }

  render() {
    const {name, homeworld, species, isFavorited} = this.props.character;

    return (
      <article className="character__cards">
        <div className="favorite-button-container">
          { !isFavorited && <img 
          onClick={this.toggleFavorited} 
          className='article__favorite-button' 
          src={favoriteImg} 
          alt='favorite button' /> }
          { isFavorited && <img 
          onClick={this.toggleFavorited} 
          className='article__favorite-button' 
          src={activeFavoriteImg} 
          alt='favorite button' /> }
        </div>
        <h5 className="character__card--element">Name: <span className="character__card--info">{name}</span></h5>
        <h5 className="character__card--element">HomeWorld: <span className="character__card--info">{homeworld[0]}</span></h5>
        <h6 className="character__card--element">Population: <span className="character__card--info">{homeworld[1]}</span></h6>
        <h5 className="character__card--element">Species: <span className="character__card--info">{species}</span></h5>
        <div>
          <h6 className="character__card--element">Featured Films:</h6>
          <div className="character__card--div">
          {this.returnFilms()}
          </div>
        </div>
      </article>
    );
  };
};

Character.propTypes = {
  character: PropTypes.object,
  name: PropTypes.string,
  homeworld: PropTypes.string,
  species: PropTypes.string,
  isFavorited: PropTypes.bool,
  updateFavorite: PropTypes.func,
  films: PropTypes.array,
  toggleFavorited: PropTypes.func,
  returnFilms: PropTypes.func
}

export default Character;

