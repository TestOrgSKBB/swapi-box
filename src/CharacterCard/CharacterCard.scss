$dark-grey: rgba(32, 37, 40, .8);
$card-background: rgba(145, 145, 145, 0.8);
$element-background: rgba(51, 51, 51, .5);
$red-font: #963025;

.character__cards {
  background: $card-background;
  border: 2px solid white;
  color: white;
  width: 26%;
  margin: 1% 1%;
  height: 57vh;
  text-align: center;
  position: relative;
  -webkit-transition: -webkit-transform .8s ease-in-out;
  transition: transform .1s ease-in-out;
}

.character__cards:hover {
  -webkit-transform: rotate(2deg);
  transform: rotate(2deg);
}

.character__card--element {
  background: $dark-grey;
  font-size: 1.5em;
  margin-top: 2%;
  margin-bottom: 2%;
}

.character__card--div {
  margin-top: 6%;
}

.character__card--films {
  font-size: 1em;
  margin-top:2%;
  margin-block-end: 0em;
  margin-block-start: 1%;
  background-color: $element-background;
  color:white;
}

.character__card--info {
  color: $red-font;
  font-size: 1.25em;
  text-shadow: 1px 0px $card-background;
}

.article__favorite-button {
  position: absolute;
  left: 0;
  max-width: 9%;
  margin-top: 2.4%;
  margin-left: 2%;
  -webkit-transition: -webkit-transform .8s ease-in-out;
  transition:         transform .3s ease-in-out;
}

.article__favorite-button:hover {
  cursor:pointer;
  // max-width: 12%;
  -webkit-transform: rotate(90deg);
  transform: rotate(90deg);
}
