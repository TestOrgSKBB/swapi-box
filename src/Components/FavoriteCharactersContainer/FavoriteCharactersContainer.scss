#favorite-characters {
  display: flex;
  flex-wrap: wrap;
  height: 80vh;
  justify-content: center;
  overflow: scroll;
}

.fav_character-title {
  color: #919192;
  font-size: 3em;
  position: absolute;
  top: 0;
}

.button__back {
  max-width: 4%;
  display: inline;
  position: absolute;
  top: 12%;
  left: 1%;
  transition: transform .2s;
}

.button__back:hover {
  transform: scale(1.1); 
  border: 1px solid #BFCAC3;
  border-radius: 4em;
}