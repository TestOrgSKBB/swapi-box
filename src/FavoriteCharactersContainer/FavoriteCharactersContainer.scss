#favorite-characters {
  display: flex;
  flex-wrap: wrap;
  height: 80vh;
  justify-content: center;
  overflow: scroll;
}

.fav_character-title {
  color: #919192;
  font-size: 3em;
  position: absolute;
  top: 0;
}