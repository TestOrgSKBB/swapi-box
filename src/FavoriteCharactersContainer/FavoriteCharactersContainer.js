import React, { Component } from 'react';
import CharacterCard from '../CharacterCard/CharacterCard';
import PropTypes from 'prop-types';
import './FavoriteCharactersContainer.scss';

class FavoriteCharactersContainer extends Component {

  displayFavoriteCharacters = () => {
    return this.props.characters.map( (character, i) => 
      <CharacterCard 
        character={character} 
        updateFavorite={this.props.updateFavorite} 
        key={i} 
      />)
  };


  render() {
    const favoriteCharacters = this.displayFavoriteCharacters();
    const hasFavChars = favoriteCharacters.length > 0;
    const display = hasFavChars ? favoriteCharacters : <h1>No Favorite Characters</h1>;
    
    return (
      <section id="favorite-characters">
        {display}
      </section>
    );
  }
}

export default FavoriteCharactersContainer;