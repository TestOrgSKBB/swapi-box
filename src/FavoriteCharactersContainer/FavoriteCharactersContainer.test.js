import React from 'react';
import FavoriteCharactersContainer from './FavoriteCharactersContainer';
import CharacterCard from '../CharacterCard/CharacterCard';

import { shallow } from 'enzyme';

describe('FavoriteCharactersContainer', () => {
  let wrapperHappy
  let wrapperSad
  const mockUpdateFavorite = jest.fn()
  let mockCharacters = [
    {
      films:['A Film', 'Another Film', 'And another'],
      homeworld: 'Earth',
      name: 'A name',
      population: '2000',
      species:'human'
    }
  ]
  let mockNoCharacters = []
  
 beforeEach(() => {
  wrapperHappy= shallow(<FavoriteCharactersContainer 
   key={100}
   characters={mockCharacters}
   updateFavorite={mockUpdateFavorite}
    />)
  wrapperSad=shallow(<FavoriteCharactersContainer 
    key={100}
    characters={mockNoCharacters}
    updateFavorite={mockUpdateFavorite}
     />)
  })

    it('should match the snapshot with the correct data', () => {
      expect(wrapperHappy).toMatchSnapshot();
    })

    it('should match the snapshot with the correct data', () => {
      expect(wrapperSad).toMatchSnapshot();
    })


 });

