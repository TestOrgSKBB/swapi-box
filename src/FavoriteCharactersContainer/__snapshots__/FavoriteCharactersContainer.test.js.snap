// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteCharactersContainer should match the snapshot with the correct data 1`] = `
<section
  id="favorite-characters"
>
  <Character
    character={
      Object {
        "films": Array [
          "A Film",
          "Another Film",
          "And another",
        ],
        "homeworld": "Earth",
        "name": "A name",
        "population": "2000",
        "species": "human",
      }
    }
    key="0"
    updateFavorite={[MockFunction]}
  />
</section>
`;

exports[`FavoriteCharactersContainer should match the snapshot with the correct data 2`] = `
<section
  id="favorite-characters"
>
  <h1>
    No Favorite Characters
  </h1>
</section>
`;
